shell/platform/darwin/macos/framework/Source/FLEExternalTextureGL.mm

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FLEExternalTextureGL.h"
+
+#import <AppKit/AppKit.h>
+#import <CoreVideo/CoreVideo.h>
+#import <OpenGL/gl.h>
+
+@implementation FLEExternalTextureGL {
+  CVOpenGLTextureCacheRef _textureCache;
+  CVPixelBufferRef _pixelBuffer;
+  id<FLETexture> _texture;
+}
+
+- (instancetype)initWithFLETexture:(id<FLETexture>)texture {
+  self = [super init];
+  if (self) {
+    _texture = texture;
+    _pixelBuffer = nil;
+    _textureCache = nil;
+  }
+  return self;
+}
+
+static void OnGLTextureRelease(CVPixelBufferRef pixelBuffer) {
+  CVPixelBufferRelease(pixelBuffer);
+}
+
+- (int64_t)textureID {
+  return reinterpret_cast<int64_t>(self);
+}
+
+- (BOOL)populateTextureWithWidth:(size_t)width
+                          height:(size_t)height
+                         texture:(FlutterOpenGLTexture*)texture {
+  if (!_texture) {
```

Comment:
Would this be a programming error? If so, it should be an assertion (in the initializer) rather than a silent no-op here.